src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidZeroLengthArrayAllocations.cs

```diff
@@ -66,13 +66,13 @@ private void AnalyzeOperation(OperationAnalysisContext context, IMethodSymbol ar
             AnalyzeOperation(context, arrayEmptyMethodSymbol, IsAttributeSyntax);
         }
 
-        private static void AnalyzeOperation(OperationAnalysisContext context, IMethodSymbol arrayEmptyMethodSymbol, Func<SyntaxNode, bool> isAttributeSytnax)
+        private static void AnalyzeOperation(OperationAnalysisContext context, IMethodSymbol arrayEmptyMethodSymbol, Func<SyntaxNode, bool> isAttributeSyntax)
         {
             IArrayCreationOperation arrayCreationExpression = (IArrayCreationOperation)context.Operation;
 
             // We can't replace array allocations in attributes, as they're persisted to metadata
             // TODO: Once we have operation walkers, we can replace this syntactic check with an operation-based check.
-            if (arrayCreationExpression.Syntax.Ancestors().Any(isAttributeSytnax))
+            if (arrayCreationExpression.Syntax.AncestorsAndSelf().Any(isAttributeSyntax))
```

Comment:
`params` parameters have the attribute itself as their syntax.

Note: we could potentially check for an `IAttributeOperation` itself, but I'm not familiar with the process for updating the reference used by the analyzer itself so it would know about that type.

Reply:
> Note: we could potentially check for an IAttributeOperation itself, but I'm not familiar with the process for updating the reference used by the analyzer itself so it would know about that type.

In case you wanted to go this route, new operations goes into Lightup layer, e.g:

https://github.com/dotnet/roslyn-analyzers/blob/793113a41d7f21b03470521bf48438f2abd9b12f/src/Utilities/Compiler/Lightup/OperationKindEx.cs#L11

https://github.com/dotnet/roslyn-analyzers/blob/793113a41d7f21b03470521bf48438f2abd9b12f/src/Utilities/Compiler/Lightup/IUsingDeclarationOperationWrapper.cs#L13

We can't yet update the product code to consume latest Roslyn versions.

Then, to consume a Roslyn version in tests that have `IAttributeOperation`, we update the following:

https://github.com/dotnet/roslyn-analyzers/blob/793113a41d7f21b03470521bf48438f2abd9b12f/eng/Versions.props#L48

Reply:
We would have to leave this check in the code anyway, so I don't think it's worth it.

Reply:
> params parameters have the attribute itself as their syntax.

Just out of curiosity is that all params parameters or just ones that are zero-length like this?

Reply:
Just the implicit ones. It matches the behavior for use in non-attribute locations (ie, just a regular invocation.